### DISCUSSION TITLE
What were some of the reasons to refactor all components to a single package in Chakra V3?

### DISCUSSION THREAD

**Discussion by endpoint-vtsang** [chakra-ui/chakra-ui#8394] [General] [answered]
Hi Chakra Team 👋 ,

I am curious on the decision to [refactor all the components to a single package](https://github.com/chakra-ui/chakra-ui/pull/8153/files) for Chakra V3.

What were some of the drawbacks in maintaining the old system?

Currently I am working on investigating packaging approaches and when I observe component libraries like [paste](https://github.com/twilio-labs/paste) or [react-spectrum](https://github.com/adobe/react-spectrum), they provide both a single version library as well individual packages for each component. This flexibly allows consumers to hold off on certain updates for a component by using a previous individual component package while updating to the major change which comes with breaking changes. 

Any insights on this, would be helpful! Thanks!

**Comment by j-vitali** [comment]
They also uses ariakit at the bottom, and then style the components....
https://ariakit.org/components

**Comment by segunadebayo** [comment, marked as answer]
To keep the installation more simple and straightforward
